frontend/packages/console-plugin-sdk/src/store.ts

```diff
@@ -31,14 +35,14 @@ export const getGatingFlagNames = (extensions: Extension[]): string[] =>
  * _For now, the runtime list of extensions is assumed to be immutable._
  */
 export class PluginStore {
-  private readonly extensions: ExtensionWithMetadata[];
+  private readonly extensions: Extension[];
 
   public readonly registry: ExtensionRegistry; // TODO(vojtech): legacy, remove
 
   public constructor(plugins: ActivePlugin[]) {
     this.extensions = _.flatMap(
       plugins.map((p) =>
-        p.extensions.map((e) => Object.freeze(augmentExtension(sanitizeExtension({ ...e }), p))),
+        p.extensions.map((e) => Object.freeze(augmentExtension(sanitizeExtension(e), p))),
```

Comment:
This change now makes the extension mutated by these actions. Is there a benefit to mutating the extension?

Reply:
`PluginStore` takes ownership of all extensions passed to it (right now via `ActivePlugin` interface).

There isn't an inherent benefit to mutating passed extension instances, this was done because `augmentExtension` computes index as `p.extensions.indexOf(e)` which doesn't work for new objects.

Anyway I think you're right so I'll update the code to keep creating new extension objects as before.

Reply:
`.map` has the index available if you prefer to provide a 3rd argument to `augmentExtensions`?

Reply:
Oh right, forgot about that. I've used `p.extensions.indexOf(e)` via closure but I can update it to use the `index` argument of `map` function.